cpp/src/arrow/array.h

```diff
@@ -798,7 +798,12 @@ class ARROW_EXPORT FixedSizeListArray : public Array {
     i += data_->offset;
     return static_cast<int32_t>(list_size_ * i);
   }
+#ifdef _MSC_VER
   int32_t value_length(int64_t i = 0) const { return list_size_; }
+#else
+  int32_t value_length(__attribute__((unused)) int64_t i = 0) const { return list_size_; }
```

Comment:
Can we use the `ARROW_ARG_UNUSED()` macro here?

Reply:
I was not very clear about arrow library, and I check your suggestion right now, and found that:
#define ARROW_ARG_UNUSED(x) 
then... I did not think it will works well in clang/g++, otherwise change the macro implementation to my code?

Reply:
emmm...
because arrow-cpp employ a very not-strictly compiling flags setting, and got too many warnings in compiling process.
Then I use -isystem arrow_src instead of -I arrow_src in my makefile.
So, if you guys didn't think unused-parameter or other warnings was big thing, ignoring my PR is ok.

Reply:
I would be surprised if this were the only place where a parameter isn't used. Also, @kou is right that this should use a macro.

Reply:
@pitrou no，very much places , so I gave it up, maybe the arrow-cpp should be implemented by more c++-like.